content/build/bake/expressions.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Expression evaluation in Bake
+description: Learn about advanced Bake features, like user-defined functions
+keywords: build, buildx, bake, buildkit, hcl, expressions, evaluation, math, arithmetic, conditionals
+aliases:
+  - /build/bake/advanced/
+---
+
+Bake files in the HCL format support expression evaluation, which lets you
+perform arithmetic operations, conditionally set values, and more.
+
+## Arithmetic operations
+
+You can perform arithmetic operations in expressions. The following example
+shows how to add two numbers together.
+
+```hcl {title=docker-bake.hcl}
+sum = 7*6
+
+target "default" {
+	args = {
+		answer = sum
+	}
+}
+```
+
+Printing the Bake file with the `--print` flag shows the evaluated value for
+the `answer` build argument.
+
+```console
+$ docker buildx bake --print app
+```
+
+```json
+{
+  "target": {
+    "default": {
+      "context": ".",
+      "dockerfile": "Dockerfile",
+      "args": {
+        "answer": "42"
+      }
+    }
+  }
+}
+```
+
+## Ternary operators
+
+You can use ternary operators to conditionally register a value.
+
+The following example adds a tag only when a variable is not empty, using the
+`notequal` function.
+
+```hcl {title=docker-bake.hcl}
+variable "TAG" {}
+
+target "default" {
+  context="."
+  dockerfile="Dockerfile"
+  tags = [
+    "my-image:latest",
+    notequal("",TAG) ? "my-image:${TAG}": "",
```

Comment:
This is smart to introduce users to functions with `notequal`. Maybe would cross link to `funcs.md`?